public/js/History.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * History: Keep track of user's sample history.
+ *
+ * This object emulates the typical back/forward behaviour in browsers.
+ *
+ * @constructor
+ */
+function History() {
+    this.sampleHistory = [];
+    this.current = -1;
+    this.lastSample = null;
+    this.length = 0;
+}
+
+/**
+ * add: Adds an item to the history.
+ *
+ * Any elements after the current iterator are removed.
+ *
+ * @this {History}
+ * @param {string} sampleId - The sampleID to add to the history.
+ */
+History.prototype.add = function(sampleId) {
+    // only add a new item to the history if it was not just added
+    if(this.lastSample !== sampleId) {
+        // add new entry to array if no more space in current history
+        if(this.current === this.sampleHistory.length-1) {
+            this.current++;
+            this.sampleHistory.push(sampleId);
+        }
+        else {
+            // otherwise overwrite old history if room in array
```

Comment:
For consistency, move this comment above the else statement

Other than that, great function, clear, compact, easy to follow! =)